addons/delivery/models/sale_order.py

```diff
@@ -23,7 +23,7 @@ def _compute_is_service_products(self):
     def _compute_amount_total_without_delivery(self):
         self.ensure_one()
         delivery_cost = sum([l.price_total for l in self.order_line if l.is_delivery])
-        return self.env['delivery.carrier']._compute_currency(self, self.amount_total - delivery_cost, 'pricelist_to_company')
+        return self.env['delivery.carrier']._compute_currency(self, super()._compute_amount_total_without_delivery() - delivery_cost, 'pricelist_to_company')
```

Comment:
There is an issue with converting the currency here:
We do not know which one of `delivery` of `sale_gift_card` will be called first, if `sale_gift_card`'s version is called first
```py
    def _compute_amount_total_without_delivery(self):
        self.ensure_one()
        # Add back 'payment' rewards from the total without delivery, they should count towards the delivery price goal.
        lines = self.order_line.filtered(lambda l: l.gift_card_id)
        return super()._compute_amount_total_without_delivery() + sum(lines.mapped('price_unit'))
```
`super()._compute_amount_total_without_delivery()` will return a converted price (for example in EURO) and we might add some prices in USD to it (if the sales order is in USD), it should be up to the caller to convert to the wanted currency.

Reply:
I want to make sure I understand, so I will limit the _compute_amount_total_without_delivery  to just return the computed float and wrap it with _compute_currency where it's needed?

Reply:
> I want to make sure I understand, so I will limit the _compute_amount_total_without_delivery to just return the computed float and wrap it with _compute_currency where it's needed?

Yes